drivers/include/extend/gpio.h

```diff
@@ -0,0 +1,491 @@
+                                                  mode, flank, cb, arg);
+    }
+#ifdef MODULE_PERIPH_GPIO
+    return gpio_cpu_init_int(pin, mode, flank, cb, arg);
+#endif
+    return -1;
+}
+
+/**
+ * @brief   Implementation by the GPIO extension interface, see @ref gpio_irq_enable
+ */
+static inline void gpio_irq_enable(gpio_t pin)
+{
+    if (GPIO_EXT_IS(pin)) {
+        assert(GPIO_EXT_PIN_DEV(pin) < ARRAY_SIZE(gpio_ext));
+        GPIO_EXT_DEV(pin).driver->irq_enable(GPIO_EXT_DEV(pin).dev,
+                                             GPIO_EXT_PIN_NUM(pin));
+        return;
+    }
+#ifdef MODULE_PERIPH_GPIO
+    gpio_cpu_irq_enable(pin);
+#endif
+}
+
+/**
+ * @brief   Implementation by the GPIO extension interface, see @ref gpio_irq_disable
+ */
+static inline void gpio_irq_disable(gpio_t pin)
+{
+    if (GPIO_EXT_IS(pin)) {
+        assert(GPIO_EXT_PIN_DEV(pin) < ARRAY_SIZE(gpio_ext));
+        GPIO_EXT_DEV(pin).driver->irq_disable(GPIO_EXT_DEV(pin).dev,
+                                              GPIO_EXT_PIN_NUM(pin));
+        return;
+    }
+#ifdef MODULE_PERIPH_GPIO
+    gpio_cpu_irq_disable(pin);
+#endif
+}
+
+#endif /* MODULE_PERIPH_GPIO_IRQ || DOXYGEN */
+
+/**
+ * @brief   Implementation by the GPIO extension interface, see @ref gpio_read
+ */
+static inline int gpio_read(gpio_t pin)
+{
+    if (GPIO_EXT_IS(pin)) {
+        assert(GPIO_EXT_PIN_DEV(pin) < ARRAY_SIZE(gpio_ext));
+        return GPIO_EXT_DEV(pin).driver->read(GPIO_EXT_DEV(pin).dev,
+                                              GPIO_EXT_PIN_NUM(pin));
+    }
+#ifdef MODULE_PERIPH_GPIO
+    return gpio_cpu_read(pin);
+#endif
+    return 0;
+}
+
+/**
+ * @brief   Implementation by the GPIO extension interface, see @ref gpio_set
+ */
+static inline void gpio_set(gpio_t pin)
+{
+    if (GPIO_EXT_IS(pin)) {
+        assert(GPIO_EXT_PIN_DEV(pin) < ARRAY_SIZE(gpio_ext));
+        GPIO_EXT_DEV(pin).driver->set(GPIO_EXT_DEV(pin).dev,
+                                      GPIO_EXT_PIN_NUM(pin));
+        return;
+    }
+#ifdef MODULE_PERIPH_GPIO
+    gpio_cpu_set(pin);
+#endif
+}
+
+/**
+ * @brief   Implementation by the GPIO extension interface, see @ref gpio_clear
+ */
+static inline void gpio_clear(gpio_t pin)
+{
+    if (GPIO_EXT_IS(pin)) {
+        assert(GPIO_EXT_PIN_DEV(pin) < ARRAY_SIZE(gpio_ext));
+        GPIO_EXT_DEV(pin).driver->clear(GPIO_EXT_DEV(pin).dev,
+                                        GPIO_EXT_PIN_NUM(pin));
+        return;
+    }
+#ifdef MODULE_PERIPH_GPIO
+    gpio_cpu_clear(pin);
+#endif
+}
+
+/**
+ * @brief   Implementation by the GPIO extension interface, see @ref gpio_toggle
+ */
+static inline void gpio_toggle(gpio_t pin)
+{
+    if (GPIO_EXT_IS(pin)) {
+        assert(GPIO_EXT_PIN_DEV(pin) < ARRAY_SIZE(gpio_ext));
+        GPIO_EXT_DEV(pin).driver->toggle(GPIO_EXT_DEV(pin).dev,
+                                         GPIO_EXT_PIN_NUM(pin));
+        return;
+    }
+#ifdef MODULE_PERIPH_GPIO
+    gpio_toggle(pin);
```

Comment:
should be `gpio_cpu_toggle(pin);`

Reply:
Ups, of course. I thought I fixed it already after a static test failed.